### PR TITLE
 publish attest cert, react to quote requests from other agents

### PR DESCRIPTION
**Pull request by cshari-zededa**
Tested that attest cert is published correctly with the proposed changes:

(ns: pillar) 4f8f3ca7-17e5-4348-8054-aad40071319b:/# cat /var/run/tpmmgr/EdgeNod
eCert/
9c091df7766a840db60b62cd4ff06fff.json  f18aaba7c3980dab814f226c91227700.json
(ns: pillar) 4f8f3ca7-17e5-4348-8054-aad40071319b:/# cat /var/run/tpmmgr/EdgeNod
eCert/*
{"HashAlgo":1,"CertID":"nAkd93ZqhA22C2LNT/Bv/w==","CertType":2,"Cert":"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRZThZYy9GWDVablVpMVdWZE9zV05nVEFLQmdncWhrak9QUVFEQWpCZ01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0F3S1EyRnNhV1p2Y201cFlURVVNQklHQTFVRUJ3d0xVMkZ1ZEdFZwpRMnhoY21FeEZEQVNCZ05WQkFvTUMxcGxaR1ZrWVN3Z1NXNWpNUkF3RGdZRFZRUUREQWR2Ym1KdllYSmtNQjRYCkRUSXdNRGN3TVRFek1Ea3lOMW9YRFRRd01EWXlOakV6TURreU4xb3dlakVMTUFrR0ExVUVCaE1DVlZNeEN6QUoKQmdOVkJBZ1RBa05CTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcGMyTnZNUjB3R3dZRFZRUUtFeFJVYUdVZwpUR2x1ZFhnZ1JtOTFibVJoZEdsdmJqRW5NQ1VHQTFVRUF4TWVSR1YyYVdObElFRjBkR1Z6ZEdGMGFXOXVJR05sCmNuUnBabWxqWVhSbE1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRWV5QkZmR1Zhd0J5M09JbHoKMU5WSHE0VFZDbjNUeUxzbGZ6NHY3N0Ftcm1saStkVWg3MEUyS203MGFyUGYwa1Z2YUlEVzNkT0xYUUYrSWttTwovb0FxSnFNMU1ETXdEZ1lEVlIwUEFRSC9CQVFEQWdXZ01CTUdBMVVkSlFRTU1Bb0dDQ3NHQVFVRkJ3TUJNQXdHCkExVWRFd0VCL3dRQ01BQXdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWdheDFlNW9wVmtIMlZGamV5VDV0bzBQUCsKWkk2UGVDZGlDSThQZ2NGSUN3QUNJUURZRWoya2FzRjhEbGs5dFpnNXRHK0xYa3RhK3FUblhudnBGTlArUmRiNgpsUT09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K","IsTpm":true}
{"HashAlgo":1,"CertID":"8Yqrp8OYDauBTyJskSJ3AA==","CertType":4,"Cert":"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNCekNDQWF5Z0F3SUJBZ0lRQmNROXVtZTM3UXhIRE5nc1J2TmVoREFLQmdncWhrak9QUVFEQWpCZ01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0F3S1EyRnNhV1p2Y201cFlURVVNQklHQTFVRUJ3d0xVMkZ1ZEdFZwpRMnhoY21FeEZEQVNCZ05WQkFvTUMxcGxaR1ZrWVN3Z1NXNWpNUkF3RGdZRFZRUUREQWR2Ym1KdllYSmtNQjRYCkRUSXdNRGN3TVRFek1Ea3lOMW9YRFRRd01EWXlOakV6TURreU4xb3djekVMTUFrR0ExVUVCaE1DVlZNeEN6QUoKQmdOVkJBZ1RBa05CTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcGMyTnZNUjB3R3dZRFZRUUtFeFJVYUdVZwpUR2x1ZFhnZ1JtOTFibVJoZEdsdmJqRWdNQjRHQTFVRUF4TVhSR1YyYVdObElFVkRSRWdnWTJWeWRHbG1hV05oCmRHVXdXVEFUQmdjcWhrak9QUUlCQmdncWhrak9QUU1CQndOQ0FBVFZibk56aHJwbENxNUZsc3lIMmdFUTNIT2kKNW5QTjBlcUd6RDZpTXFQV0I4UjdGOTF5bDNUN1h2RzgvM3J4ZTljalNyeWVFcG5QcmtRek1pUS80bWZqb3pVdwpNekFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJS3dZQkJRVUhBd0V3REFZRFZSMFRBUUgvCkJBSXdBREFLQmdncWhrak9QUVFEQWdOSkFEQkdBaUVBeXd3Nit0Y01rQzVxN3NYVWlsN0h6M3dyakJNYlBOU0UKYVJDOVdwckcrd3NDSVFDM0g4dDNEZjExVjd3ZVR1b2xJbm1aNHJVdktMTmVKUWJTU29zTHRRNzM2Zz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K","IsTpm":true}(ns: pillar) 4f8f3ca7-17e5-4348-8054-aad40071319b:/# 

Signed-off-by: Hariharasubramanian C S <cshari@zededa.com>

### CODE REVIEW

File: pkg/pillar/cmd/tpmmgr/tpmmgr.go [hunk 1]
```diff
@@ -631,32 +632,61 @@ func printCapability() {
...
-		return
+		return nil, nil, nil, err
 	}
 	defer rw.Close()
-	for i := 0; i < 23; i++ {
-		pcrVal, err := tpm2.ReadPCR(rw, i, tpm2.AlgSHA256)
+	pcrs := make([]types.PCRValue, 0)
+
+	var i uint8
+	for i = 0; i < 15; i++ {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/tpmmgr/tpmmgr.go:649]: Why the hardcoded max 15? Is that the max in the TCG specification? Would make sense to define as a CONST in any case.

Comment (cshari-zededa) [reply pkg/pillar/cmd/tpmmgr/tpmmgr.go]: addressed it, changed it (to 0-23 per TCG spec), and made it a constant.

**Comment by cshari-zededa** [comment]
> LGTM
Please check if this can be pulled, as per the latest changes.

**Comment by rvs** [comment]
This looks like it is ready to be merged @cshari-zededa -- is there anything left?

**Comment by cshari-zededa** [comment]
> This looks like it is ready to be merged @cshari-zededa -- is there anything left?

This is ready to be pulled @rvs

**Pull request merged**